core/iwasm/interpreter/wasm_loader.c

```diff
@@ -4054,17 +4062,47 @@ load_global_section(const uint8 *buf, const uint8 *buf_end, WASMModule *module,
     return false;
 }
 
+static bool
+check_duplicate_exports(WASMModule *module, char *error_buf,
+                        uint32 error_buf_size)
+{
+    char **names = loader_malloc(module->export_count * sizeof(char *),
+                                 error_buf, error_buf_size);
+    bool result = false;
+
+    if (!names) {
+        return result;
+    }
+
+    for (uint32 i = 0; i < module->export_count; i++) {
```

Comment:
had better declare `uint32 i` at the beginning of this function, we usually do that way since here it is C source code.

Reply:
curious, variable declarations can occur other than at the top of blocks since C99. Other parts of the code make use of C99 features, e.g. a designated initializer in `core/iwasm/aot/aot_runtime.c`. What is the reason that you stick to this style?

Reply:
Yes, declaring variables in the middle of a function is good, I just think that declaring a variable in a for loop is not a good coding style in C, it is declared not so explicitly and may make other variables be shadowed. And a little afraid that some compilers may report warning or error.

Reply:
Fixed